jdbc-dao/src/main/java/com/bobocode/dao/CompanyDaoImpl.java

```diff
@@ -0,0 +1,96 @@
+    }
+
+    @Override
+    public void save(Company company) {
+        try (Connection connection = dataSource.getConnection()) {
+            PreparedStatement insertStatement = createStatement(connection, company);
+            executeInsert(insertStatement);
+            Long id = fetchGeneratedId(insertStatement);
+            company.setId(id);
+            System.out.println("Company was created! Id: " + id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private Long fetchGeneratedId(PreparedStatement insertStatement) throws SQLException {
+        ResultSet generatedKeys = insertStatement.getGeneratedKeys();
+        if (generatedKeys.next()) {
+            return generatedKeys.getLong(1);
+        } else {
+            throw new SQLException("Can not obtain an account ID");
+        }
+    }
+
+    private void executeInsert(PreparedStatement insertStatement) throws SQLException {
+        int i = insertStatement.executeUpdate();
+        if (i == 0) {
+            throw new DaoOperationException("Company was not created");
+        }
+    }
+
+    private PreparedStatement createStatement(Connection connection, Company company) throws SQLException {
+        PreparedStatement preparedStatement = connection.prepareStatement(
+                "INSERT INTO company(name, phone) VALUES(?,?);",
+                PreparedStatement.RETURN_GENERATED_KEYS
+        );
+        preparedStatement.setString(1, company.getName());
+        preparedStatement.setString(2, company.getPhone());
+        return preparedStatement;
+    }
+
+    @Override
+    public List<Company> findAll() {
+        try (Connection connection = dataSource.getConnection()) {
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery("SELECT * FROM company;");
+            return collectToList(resultSet);
+        } catch (SQLException e) {
+            throw new DaoOperationException(e.getMessage());
+        }
+    }
+
+    // TODO: 21.08.17 create this method for all model
+    private List<Company> collectToList(ResultSet resultSet) throws SQLException {
+        List<Company> companies = new ArrayList<>();
+        while (resultSet.next()) {
+            Company company = parseRow(resultSet);
+            companies.add(company);
+        }
+        return companies;
+    }
+
+    private Company parseRow(ResultSet resultSet) throws SQLException {
+        return new Company() {{
+            setId(resultSet.getLong(1));
+            setName(resultSet.getString(2));
+            setPhone(resultSet.getString(3));
+        }};
+    }
+
+    @Override
+    public List<Company> findByName(String name) {
+        try (Connection connection = dataSource.getConnection()) {
+            return collectToList(connection
+                    .createStatement()
+                    .executeQuery("SELECT * FROM company WHERE name='" + name + "';"));
```

Comment:
Again, it's better to use parameters. In addition, this piece of code is very readable